sparsecoding/data/datasets/datasets.py

```diff
@@ -27,29 +27,26 @@ def __init__(self, img_dir, patch_size, patch_overlap, data_key='IMAGESr',device
         self.device = torch.device("cpu") if device is None else device
         self.patch_size = patch_size
         self.patch_overlap = patch_overlap
-        
+
         # load image dataset
         images_dict = sio.loadmat(img_dir)
-        self.images = np.asarray(images_dict[data_key]) # scalar - (m,m,n_img)
-        
+        self.images = np.asarray(images_dict[data_key])  # scalar - (m,m,n_img)
+
         # make patches
         self.extractpatches()
-    
-    
+
     def __len__(self):
         '''
         return number of patches
         '''
         return self.patches.shape[0]
 
-    
     def __getitem__(self, idx):
         '''
         return patch at idx
         '''
-        return self.patches[idx,:]
-    
-    
+        return self.patches[idx, :]
+
     def extractpatches(self):
```

Comment:
This function will be generally useful beyond just NaturalScenes. We can go ahead and include this in the merge, but I think we should instead put this into a data utils module, then import it for use here rather than defining it within the NaturalScenes class.

Reply:
Also, the name should be snake case:
def extract_patches